app/src/main/java/com/mithridat/nonoconverter/backend/nonogram/Field.java

```diff
@@ -0,0 +1,324 @@
+                public Field[] newArray(int size) {
+                    return new Field[size];
+                }
+
+            };
+
+    /**
+     * Method for getting number of field rows
+     *
+     * @return number of field rows
+     */
+    public int getRows() {
+        return _rows;
+    }
+
+    /**
+     * Method for getting number of field columns
+     *
+     * @return number of field columns
+     */
+    public int getCols() {
+        return _cols;
+    }
+
+    /**
+     * Method for getting length of _rows or _cols
+     *
+     * @param type - ROW, if _rows
+     *               COL, if _cols
+     * @return length of _rows or _cols
+     */
+    public int getLength(int type) {
+        if (type == ROW) return getRows();
+        return getCols();
+    }
+
+
+    /**
+     * Method for getting field cell color with coordinates (i,j)
+     *
+     * @param i - field cell row number
+     * @param j - field cell column number
+     * @return field cell color with coordinates (i,j)
+     */
+    public int getColor(int i, int j) {
+        return _field[i][j];
+    }
+
+    /**
+     * Method for getting field cell color with coordinates (i,j) or (j, i)
+     *
+     * @param i - i
+     * @param j - j
+     * @param type - ROW, if (i, j)
+     *               COL, if (j, i)
+     * @return field cell color with coordinates (i,j) or (j, i)
+     */
+    public int getColor(int i, int j, int type) {
+        if (type == ROW) return getColor(i, j);
+        return getColor(j, i);
+    }
+
+
+    /**
+     * Method for setting field cell color with coordinates (i,j)
+     *
+     * @param i - field cell row number
+     * @param j - field cell column number
+     * @param color - color
+     */
+    public void setColor(int i, int j, int color) {
+        _field[i][j] = color;
+    }
+
+    /**
+     * Method for setting field cell color with coordinates (i,j) or (j, i)
+     *
+     * @param i - i
+     * @param j - j
+     * @param color - color
+     * @param type - ROW, if (i, j)
+     *               COL, if (j, i)
+     */
+    public void setColor(int i, int j, int color, int type) {
+        if (type == ROW) setColor(i, j, color);
+        setColor(j, i, color);
+    }
+
+    /**
+     * Method for getting count of colors
+     *
+     * @return count of colors
+     */
+   public int getColors() {
+        return _colorsCount;
+    }
+
+    /**
+     * Method for getting index of first cell with another color
+     * (current color - the color of cell with index = pos) in row or column
+     *
+     * @param ind - index of row or column
+     * @param pos - position, the search begins with the next cell
+     * @param dir - direction, +1 or -1
+     * @param type - ROW, if row
+     *               COL, if column
+     * @return index of first cell with another color in row or column
+     */
+    public int getAnotherColorIndex(int ind, int pos, int dir, int type) {
+        int i = pos + 1;
```

Comment:
Now it can be replaced by
```java
        int i = pos + 1, len = getLength(type);
        for (; i >= 0 && i < len; i += dir) {
            if (getColor(ind, i, type) != getColor(ind, pos, type)) break;
        }
        return i;
```